R/zchunk_L123.bcoc_awb_R_S_T_Y.R

```diff
@@ -30,43 +30,78 @@ module_emissions_L123.bcoc_awb_R_S_T_Y_DISABLED <- function(command, ...) {
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
     L103.ag_Prod_Mt_R_C_Y_GLU <- get_data(all_data, "L103.ag_Prod_Mt_R_C_Y_GLU")
-    L121.AWBshare_R_C_Y_GLU <- get_data(all_data, "L121.AWBshare_R_C_Y_GLU")
+    L121.AWBshare_R_C_Y_GLU <- get_data(all_data, "temp-data-inject/L121.AWBshare_R_C_Y_GLU")
     RCP_BC_2000 <- get_data(all_data, "emissions/RCP_BC_2000")
     RCP_OC_2000 <- get_data(all_data, "emissions/RCP_OC_2000")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` and/or 'match' calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Merge-and-Match
-    # ===================================================
+    # Extract 2000 emissions data for AWB emissions and assign emission species identifier
+    RCP_BC_2000 %>%
+      select(iso, awb) %>%
+      mutate(Non.CO2 = "BC_AWB") -> BC_AWB_2000
+
+    RCP_OC_2000 %>%
+      select(iso, awb) %>%
+      mutate(Non.CO2 = "OC_AWB") -> OC_AWB_2000
+
+    # Aggregate BC and OC emissions to GCAM regions and convert to Tg
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+      # Old system used match and ended up with just the first value instead of summing all values
+      bind_rows(OC_AWB_2000, BC_AWB_2000) %>%
+      left_join(iso_GCAM_regID, by = "iso") %>% #Not all iso's will necesarilly have an input value. This is ok.
+      group_by(Non.CO2, GCAM_region_ID) %>%
+      summarise( awb = first( awb ) )  %>%
+      filter( !is.na(awb) ) %>%
+      mutate( awb = awb * CONV_KG_TO_TG ) %>%
+      ungroup() ->
+      BCOC_AWB_2000
+    } else {
+        bind_rows(OC_AWB_2000, BC_AWB_2000) %>%
+        left_join(iso_GCAM_regID, by = "iso") %>% #Not all iso's will necesarilly have an input value. This is ok.
+        group_by(Non.CO2, GCAM_region_ID) %>%
+        summarize_if(is.numeric , sum) %>%
+        filter( !is.na(awb) ) %>%
+        mutate( awb = awb * CONV_KG_TO_TG ) %>%
+        ungroup() ->
+        BCOC_AWB_2000
+   }
+
+    # Extract only 2000 data from ag production data
+    filter( L103.ag_Prod_Mt_R_C_Y_GLU, year == 2000 ) %>%
+      select( -year ) %>%
+      mutate(GLU = as.character(GLU), GCAM_commodity = as.character(GCAM_commodity) ) ->
+      L103.ag_Prod_Mt_R_C_Y_GLU_2000
+
+    # Allocate aggreate regional year 2000 AWB emissions into GCAM crop and GLU
+    # This is done using shares from L121.AWBshare_R_C_Y_GLU, which is calculated in another chunk
+    filter(L121.AWBshare_R_C_Y_GLU, year == "X2000")  %>%
```

Comment:
Ah...here's a significant change. This is currently `temp-data-inject/` data, i.e. temporarily spliced in from the old data system, but it _will_ be coming in from another chunk in long form. So won't have an "X2000" field name.